azurelinuxagent/ga/exthandlers.py

```diff
@@ -374,11 +373,16 @@ def _extension_processing_allowed(self):
         return True
 
     def handle_ext_handlers(self, etag=None):
-        if self.ext_handlers.extHandlers is None or \
-                len(self.ext_handlers.extHandlers) == 0: # pylint: disable=len-as-condition
+        if not self.ext_handlers.extHandlers:
             logger.verbose("No extension handler config found")
             return
 
+        if self.last_etag == etag:
+            # Skip processing if GoalState incarnation did not change
+            return
+
+        logger.info("New incarnation detected, starting processing of GoalState with incarnation: {0}".format(etag))
```

Comment:
Let's rephrase this message and make it look symmetric to

 ` logger.info('ProcessGoalState completed [incarnation {0}; {1} ms]',`

Reply:
Good point, done

Reply:
Done